tests/test_function.py

```diff
@@ -665,6 +665,21 @@ def test_replace(self):
     self.assertEqual(ns.foo, function.zeros([2,3]))
     self.assertEqual(ns.default_geometry_name, 'y')
 
+  def test_pickle(self):
+    orig = function.Namespace()
+    domain, geom = mesh.unitsquare(2, 'square')
+    orig.x = geom
+    orig.v = domain.basis('std', 1)
+    orig.u = 'v_n ?lhs_n'
+    orig.f = 'cosh(x_0)'
+    pickled = pickle.loads(pickle.dumps(orig))
+    for attr in ('x', 'v', 'u', 'f'):
+      self.assertEqual(getattr(pickled, attr), getattr(orig, attr))
+
+  def test_pickle_default_geometry_name(self):
+    orig = function.Namespace()
+    pickled = pickle.loads(pickle.dumps(orig))
+    self.assertEqual(pickled.default_geometry_name, orig.default_geometry_name)
```

Comment:
Can you pick a non-default geometry name here? For example

    orig = function.Namespace(default_geometry_name='foo')

Reply:
Thanks for the review, cf forthcoming commit

Reply:
Could you move the `default_geometry_name=foo` bit to the `test_pickle_default_geometry_name` function? In the other function the value of `pickle.default_geometry_name` is not checked. You can revert the `default_geometry_name` changes in `test_pickle`. Sorry if this was not clear.